kubernetes/cluster_deploy.go

```diff
@@ -0,0 +1,148 @@
+package kubernetes
+
+import (
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+
+	"github.com/mattermost/mattermost-load-test/ltops"
+)
+
+type ChartConfig struct {
+	Global   *GlobalConfig   `yaml:"global"`
+	MySQLHA  *MySQLHAConfig  `yaml:"mysqlha"`
+	App      *AppConfig      `yaml:"mattermost-app"`
+	Loadtest *LoadtestConfig `yaml:"mattermost-loadtest"`
+}
+
+type GlobalConfig struct {
+	SiteURL           string          `yaml:"siteUrl"`
+	MattermostLicense string          `yaml:"mattermostLicense"`
+	Features          *FeaturesConfig `yaml:"features"`
+}
+
+type FeaturesConfig struct {
+	LoadTest *LoadTestFeature `yaml:"loadTest"`
+	Grafana  *GrafanFeature   `yaml:"grafana"`
+}
+
+type LoadTestFeature struct {
+	Enabled bool `yaml:"enabled"`
+}
+
+type GrafanFeature struct {
+	Enabled bool `yaml:"enabled"`
+}
+
+type MySQLHAConfig struct {
+	Enabled bool            `yaml:"enabled"`
+	Options *MySQLHAOptions `yaml:"mysqlha"`
+}
+
+type MySQLHAOptions struct {
+	ReplicaCount int `yaml:"replicaCount"`
+}
+
+type AppConfig struct {
+	ReplicaCount int           `yaml:"replicaCount"`
+	Image        *ImageSetting `yaml:"image"`
+}
+
+type LoadtestConfig struct {
+	ReplicaCount int `yaml:"replicaCount"`
+}
+
+type ImageSetting struct {
+	Tag string `yaml:"tag"`
+}
+
+func (c *Cluster) DeployMattermost(mattermostFile string, licenceFileLocation string) error {
+	log.Info("installing mattermost helm chart...")
+
+	if len(c.ReleaseName) > 0 {
+		log.Info("already installed as release '" + c.ReleaseName + "'")
+		return nil
+	}
+
+	license, err := ltops.GetFileOrURL(licenceFileLocation)
+	if err != nil {
+		return err
+	}
+
+	config := &ChartConfig{
+		Global: &GlobalConfig{
+			SiteURL:           "http://localhost:8065",
+			MattermostLicense: string(license),
+			Features: &FeaturesConfig{
+				&LoadTestFeature{Enabled: true},
+				&GrafanFeature{Enabled: true},
+			},
+		},
+		MySQLHA: &MySQLHAConfig{
+			Enabled: true,
+			Options: &MySQLHAOptions{
+				ReplicaCount: c.Configuration().DBInstanceCount,
+			},
+		},
+		App: &AppConfig{
+			ReplicaCount: c.Configuration().AppInstanceCount,
+			Image: &ImageSetting{
+				Tag: "4.9.2",
```

Comment:
Hardcoded to 4.9.2 for now so it's not running off master. When we create pipelines/jenkins jobs to create the docker images for release versions then we can update this to be a parameter